drivers/usb/device/usb_dc_smartbond.c

```diff
@@ -892,34 +917,80 @@ static void usb_dc_smartbond_isr(void)
 	}
 }
 
-static void usb_dc_smartbond_vbus_changed(bool present)
+/**
+ * USB functionality can be disabled from HOST and DEVICE side.
+ * Host side is indicated by VBUS line.
+ * Device side is decided by pair of calls usb_dc_attach()/usb_dc_detach,
+ * USB will only work when application calls usb_dc_attach() and VBUS is present.
+ * When both conditions are not met USB clock (PLL) is released, and peripheral
+ * remain in reset state.
+ */
+static void usb_change_state(bool attached, bool vbus_present)
 {
-	if (dev_state.vbus_present == present) {
+	if (dev_state.attached == attached && dev_state.vbus_present == vbus_present) {
 		return;
 	}
 
-	dev_state.vbus_present = present;
-
-	if (present) {
-		/* TODO: Add PD acquire when available */
-		/* If power event happened before USB started, delay dcd_connect
-		 * until dcd_init is called.
+	if (attached && vbus_present) {
+		dev_state.attached = true;
+		dev_state.vbus_present = true;
+#if defined(CONFIG_PM)
```

Comment:
It looks like pm_policy_state_lock_get() can always build, is `#if defined(CONFIG_PM)` required here and below?

Reply:
Thanks for advice, suggestion applied